modules/exploits/multi/script/web_delivery.rb

```diff
@@ -111,13 +106,13 @@ def initialize(info = {})
 
   register_advanced_options(
     [
-      OptBool.new('PSH-AmsiBypass',       [ true,  'PSH - Request AMSI/SBL bypass before the stager', true ]),
-      OptString.new('PSH-AmsiBypassURI',  [ false, 'PSH - The URL to use for the AMSI/SBL bypass (Will be random if left blank)', '' ]),
-      OptBool.new('PSH-EncodedCommand',   [ true,  'PSH - Use -EncodedCommand for web_delivery launcher', true ]),
-      OptBool.new('PSH-ForceTLS12',       [ true,  'PSH - Force use of TLS v1.2', true ]),
-      OptBool.new('PSH-Proxy',            [ true,  'PSH - Use the system proxy', true ]),
-      OptString.new('PSHBinary-PATH',     [ false, 'PSH (Binary) - The folder to store the file on the target machine (Will be %TEMP% if left blank)', '' ]),
-      OptString.new('PSHBinary-FILENAME', [ false, 'PSH (Binary) - The filename to use (Will be random if left blank)', '' ]),
+      OptBool.new('PSH-AmsiBypass',       [ true,  'PSH - Request AMSI/SBL bypass before the stager', true ], conditions: ['TARGET', 'in', ['PSH', 'PSH (Binary)']]),
```

Comment:
Using a more method-oriented API could transform the conditions here like this:

```suggestion
      OptBool.new('PSH-AmsiBypass',       [ true,  'PSH - Request AMSI/SBL bypass before the stager', true ], conditions: OptCondition.new('TARGET').in('PSH', 'PSH (Binary)'),
```